sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetFileFormat.scala

```diff
@@ -205,11 +212,21 @@ class ParquetFileFormat
 
       val sharedConf = broadcastedHadoopConf.value.value
 
-      lazy val footerFileMetaData =
-        ParquetFooterReader.readFooter(sharedConf, filePath, SKIP_ROW_GROUPS).getFileMetaData
+      val fileRange = HadoopReadOptions.builder(sharedConf, split.getPath)
```

Comment:
can we add these in `ParquetFooterReader`? we may need to use `try-with-resources` clause to make sure resources are properly closed.

we can just obtain the footer here and use it later for `footerFileMetaData` and pass it to `VectorizedParquetRecordReader`

Reply:
Yes, before I created and passed `ParquetFileReader` because I wanted to create one less `file.newStream()` in it (if  there is no filter pushdown), but it doesn’t seem to make much sense. I have changed to pass footer here.
Please take a look, thank you!